surface/model_openapiv3.go

```diff
@@ -439,3 +436,34 @@ func (b *OpenAPI3Builder) buildFromSchema(name string, schema *openapiv3.Schema)
 	log.Printf("Unimplemented: could not find field info for schema: %v", schema)
 	return nil
 }
+
+// buildFromOneOfAnyOfAndAllOf adds appropriate fields to the 'schemaType' given a new 'schemaOrRef'.
+func (b *OpenAPI3Builder) buildFromOneOfAnyOfAndAllOf(schemaOrRef *openapiv3.SchemaOrReference, schemaType *Type) {
+	// Related: https://github.com/googleapis/gnostic-grpc/issues/22
+	if schema := schemaOrRef.GetSchema(); schema != nil {
+		fieldInfo := b.buildFromSchemaOrReference("ATemporaryTypeThatWillBeRemoved", schemaOrRef)
+		if t := findType(b.model.Types, "ATemporaryTypeThatWillBeRemoved"); t != nil {
```

Comment:
Could you add a comment with some explanation of what is happening here? Is "ATemporaryType..." being used to collect fields from the OneOf/AnyOf/AllOfs and then "WillBeRemoved" when these fields are merged into the main type?

Reply:
Yes, exactly.